drivers/modem/ublox-sara-r4.c

```diff
@@ -95,7 +94,8 @@ static struct modem_pin modem_pins[] = {
 #define MDM_IMSI_LENGTH			16
 #define MDM_APN_LENGTH			32
 
-#define RSSI_TIMEOUT_SECS		30
+#define MODEM_RSSI_WORK_ENABLE		CONFIG_MODEM_UBLOX_SARA_RSSI_WORK
```

Comment:
Is this really working as expected? If `CONFIG_MODEM_UBLOX_SARA_RSSI_WORK` is not set, then what is the value of `MODEM_RSSI_WORK_ENABLE` in that case?
Better would be to write this as
```
#define MODEM_RSSI_WORK_ENABLE    IS_ENABLED(CONFIG_MODEM_UBLOX_SARA_RSSI_WORK)
```
then the `MODEM_RSSI_WORK_ENABLE` will have value of 0 or 1 and the checks below work as expected too.

Reply:
There are many places in zephyr with similar code:
```
git grep "#if CONFIG_" | wc -l
442
```
In fact less as there are not always boolean kconfig, but still it does work.

However, I can definitively change it if it's not the recommended way to do it ?

In this code I see different variations using directly the Kconfig with no define alias:
```
#if defined(CONFIG_MODEM_UBLOX_SARA_AUTODETECT_VARIANT)
#ifdef CONFIG_DNS_RESOLVER
```
Or I could go with your version.

@jukkar which version is the recommended one?

Reply:
The `#if defined()` is usually the preferred one as then you can easily add conditions to the checks.

Reply:
Done :)